crates/credentials/src/presentation_definition_v2.rs

```diff
@@ -0,0 +1,230 @@
+use jsonschema::{Draft, JSONSchema};
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use serde_with::skip_serializing_none;
+use std::collections::HashMap;
+
+/// Presentation Exchange
+///
+/// Presentation Exchange specification codifies a Presentation Definition data format Verifiers
+/// can use to articulate proof requirements, and a Presentation Submission data format Holders can
+/// use to describe proofs submitted in accordance with them.
+///
+/// See [Presentation Definition](https://identity.foundation/presentation-exchange/#presentation-definition)
+/// for more information.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct PresentationDefinitionV2 {
+    pub id: String,
+    pub name: Option<String>,
+    pub purpose: Option<String>,
+    pub format: Option<Format>,
+    pub submission_requirements: Option<Vec<SubmissionRequirement>>,
+    pub input_descriptors: Vec<InputDescriptorV2>,
+    pub frame: Option<HashMap<String, JsonValue>>,
+}
+
+/// Represents an input descriptor in a presentation definition.
+///
+/// See [Input Descriptor](https://identity.foundation/presentation-exchange/#input-descriptor-object)
+/// for more information.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct InputDescriptorV2 {
+    pub id: String,
+    pub name: Option<String>,
+    pub purpose: Option<String>,
+    pub format: Option<Format>,
+    pub constraints: ConstraintsV2,
+}
+
+/// Represents constraints for an input descriptor.
+///
+/// See 'constraints object' defined in
+/// [Input Descriptor](https://identity.foundation/presentation-exchange/#input-descriptor-object)
+/// for more information.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct ConstraintsV2 {
```

Comment:
nit: do we have to suffix with V2 here and elsewhere? I don't believe we have any need/desire to support v1

Reply:
I'm not opposed, but I do think it helps with clarity. There are two distinct versions of PresentationDefinition, and if v2 was omitted, I'd think this was v1 at first glance.

We currently have the "V2" suffix in the spec for offerings here: https://github.com/TBD54566975/tbdex/tree/main/specs/protocol#offering

The  Kotlin & JS codebases also have the "V2" suffix as well.

If we do decide to remove, we should probably align on that across our codebases & documentation.

Reply:
What about namespacing (directory or module)? That would be my vote across all codebases.

Reply:
That's a good idea. I'll update this to namespace these into a `v2` module. Then we can drop the suffix!

Reply:
Updated to remove the `V2` suffixes, and placed it within a `v2` module: 
https://github.com/TBD54566975/web5-rs/pull/18/commits/1ddeb54df94706146055f2a883b1b7cb39a2853a

Now consumers can use this in their code with the following use statement:
`use credentials::pex::v2::PresentationDefinition;`